src/conventional/sonde_bufr2ioda.py

```diff
@@ -430,181 +415,309 @@ def read_bufr_message(f, count, start_pos, data):
+        meta_data['stationWMO'] = np.full(target_number, string_missing_value, dtype='<S5')
+        for n, block in enumerate(meta_data['wmoBlockNumber']):
+            number = meta_data['wmoStationNumber'][n]
+            if (block > 0 and block < 100 and number > 0 and number < 1000):
+                meta_data['stationWMO'][n] = "{:02d}".format(block) + "{:03d}".format(number)
+            if n == 0:
+                count[3] += 1
+                logging.info(f"Processing sonde for station: {meta_data['stationWMO'][n]}")
+
+        # Very odd, sometimes the first level of data has some variables set to zero. Reset to missing.
+        if (meta_data['geopotentialHeight'][0] == 0 or meta_data['pressure'][0] == 0):
+            meta_data['geopotentialHeight'][0] = float_missing_value
+            meta_data['pressure'][0] = float_missing_value
+
+        # And now processing the observed variables we care about.
+        nbad = 0
+        for variable in raw_obsvars:
+            vals[variable] = np.full(target_number, float_missing_value)
+            if temp_data[variable] is not None:
+                try:
+                    vals[variable] = temp_data[variable][b:e]
+                    if len(vals[variable]) < target_number:
+                        nbad += 1
+                        logging.warning(f" var {variable} has {len(vals[variable])} "
+                                        f"elements while expecting {target_number}")
+                        vals[variable] = np.full(target_number, float_missing_value)
+                except Exception:
+                    logging.warning(f"Unable to copy {variable} data, "
+                                    f"either index [{b},{e}] must be out of range.")
+            else:
+                nbad += 1
+
+        if nbad == len(raw_obsvars):
+            logging.warning(f"No usable data in this ob, skipping it.")
+            count[2] += target_number
+            return data, count, start_pos
+
+        count[1] += target_number
+
+        # Finally transfer the meta_data to the output array.
+        for key in meta_keys:
+            data[key] = np.append(data[key], meta_data[key])
+
+        # Need to transform some variables to others (wind speed/direction to components for example).
+        uwnd = np.full(target_number, float_missing_value)
+        vwnd = np.full(target_number, float_missing_value)
+        for n, wdir in enumerate(vals['windDirection']):
+            wspd = vals['windSpeed'][n]
+            if wdir and wspd:
+                if (wdir >= 0 and wdir <= 360 and wspd >= 0 and wspd < 300):
+                    uwnd[n], vwnd[n] = met_utils.dir_speed_2_uv(wdir, wspd)
+
+        spfh = np.full(target_number, float_missing_value)
+        for n, dewpoint in enumerate(vals['dewpointTemperature']):
+            pres = meta_data['pressure'][n]
+            if dewpoint and pres:
+                if (dewpoint > 50 and dewpoint < 325 and pres > 100 and pres < 109900):
```

Comment:
Is it the case that the met_utils.specific_humidity requires these ranges? If so, please put in a comment explaining this. Thanks!